spec/lib/miq_ldap_spec.rb

```diff
@@ -119,4 +119,20 @@
     data = "\001\005\000\000\000\000\000\005\025\000\000\000+\206\301\364y\307\r\302=\336p\216\237\004\000\000"
     expect(MiqLdap.sid_to_s(data)).to eq("S-1-5-21-4106323499-3255682937-2389761597-1183")
   end
+
+  it 'returns a hostname when a hostname is availble and does not disable verify mode' do
+    allow(TCPSocket).to receive(:gethostbyname).and_return(["testhostname", "aliases", "type", "192.168.252.20"])
+    allow(TCPSocket).to receive(:new)
+    ldap = MiqLdap.new(:host => ["testhostname", "localhost", "dummy", @host])
+    expect(ldap.ldap.host).to eq("testhostname")
+    expect(ldap.ldap.inspect).to_not include("@encryption={:tls_options=>{:verify_mode=>0}}")
+  end
+
+  it 'returns an IPAddress and disables verify mode when only an IPAddress is availble' do
+    expect(TCPSocket).not_to receive(:gethostbyname)
+    allow(TCPSocket).to receive(:new)
+    ldap = MiqLdap.new(:host => ["192.168.254.15", "localhost", "dummy", @host])
+    expect(ldap.ldap.host).to eq("192.168.254.15")
+    expect(ldap.ldap.inspect).to include("@encryption={:tls_options=>{:verify_mode=>0}}")
```

Comment:
Would it be more reliable to do the following?
```ruby
expect(ldap.ldap.instance_variable_get(:@encryption)).to eq(:tls_options=>{:verify_mode=>0})
```

Reply:
@bdunne and @Fryguy would it be better to do as @bdunne suggests or todo:

```ruby
expect(ldap.ldap.instance_variable_get(:@encryption).try(:has_key_path?, :tls_options, :verify_mode)).to be_truthy
```

Reply:
I think it depends on what you're trying to test.  if you care that the value of `:verify_mode` is `0` then you want what I suggested, but if you only care that the `:verify_mode` key exists, then `:has_key_path?` is good enough.